homeassistant/components/fritz/sensor.py

```diff
@@ -322,6 +340,7 @@ def update(self) -> None:
             self._attr_available = False
             return
 
-        self._attr_native_value = self._last_device_value = self._state_provider(
-            status, self._last_device_value
-        )
+        if self._state_provider:
```

Comment:
```suggestion
        if self._state_provider is not None:
```
but when it is fine for `_state_provider` to be none?

Reply:
Good point, switched to `assert`